### PR TITLE
Always try to get CMap, even if name is not recognized

### LINKED ISSUE #391: New versions of pdfminer.six cannot extract chinese characters from pdf
**Describe the bug:**

I have observed that latest version of pdfminer.six i.e. 20200124 is not able to read Chinese character.

The version appropriately working right now is 20181108. After that all the later versions are having that problem. 

**To Reproduce:**

Simply run     `pdf2txt.py input.pdf`

**Current output:**
```
Ysj Lvf
Tel:
Fax:
Email:
...
...
Abc Lvf
28H abcdefg Road#06-90 Tower 9 The Abcdef@AbcdefGhijklmnop 8336661
```

**Expected Output:**

```
Ysj退Lvf送
Tel:
Fax:
Email:
...
...
Abc退Lvf送
28H abcdefg Road#06-90 Tower 9 The Abcdef@AbcdefGhijklmnop 8336661

```
**Reference PDF:**
[Sample pdf for chinese character.pdf](https://github.com/pdfminer/pdfminer.six/files/4352872/uncompressedunbroken.pdf)

P.S. The acrobat may show error while opening this pdf, you can ignore that error. It is prompting this error because some part of the pdf had customer data, so I have manually changed it.

### PR DESCRIPTION
**Pull request by pietermarsman**
**Pull request**

Fixes #391 

This PR allows always tries to get a `cmap_name`. Previously this was only done for . It does not break any existing behavior. 

**How Has This Been Tested?**

- By testing it on the pdf in #391 
- By adding a test that fails without this PR

**Checklist**

- [x] I have added tests that prove my fix is effective or that my feature 
  works
- [x] I have added docstrings to newly created methods and classes
- [x] I have optimized the code at least one time after creating the initial 
  version
- [x] I have updated the [README.md](../README.md) or I am verified that this
  is not necessary
- [x] I have updated the [readthedocs](../docs/source) documentation or I 
  verified that this is not necessary
- [x] I have added a consice human-readable description of the change to 
  [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

**Comment by pietermarsman** [comment]
@fakabbir Could you elaborate on that? I don't see the difference in the `cmap_name` between before and after.

**Comment by fakabbir** [comment]
So, at this place https://github.com/pdfminer/pdfminer.six/pull/438/files#diff-9d138ff43c58cd4903b1e16ce49c98fcR766  `cmap_name = IDENTITY_ENCODER.get(cmap_name, cmap_name)` you have ensured that the the cmap_name is always assigned in case even if the name is not in IDENTITY_ENCODER. I see this making a difference.

**Comment by pietermarsman** [comment]
But in the old code the `cmap_name` is also always assigned, right? It is set to `"unknown"` if it cannot be inferred from the pdf. And that did not change. 

Happy to improve thing here, but I'm not (yet) seeing it.

**Comment by fakabbir** [comment]
> But in the old code the `cmap_name` is also always assigned, right? It is set to `"unknown"` if it cannot be inferred from the pdf. And that did not change.
> 
> Happy to improve thing here, but I'm not (yet) seeing it.

I felt it's getting assigned as `UniGB-UCS2-H` instead of `unknown`. But if that not the case, I wonder why solved the issue. Let me try debug the test case.

**Comment by pietermarsman** [comment]
@fakabbir I'm a bit confused on what to do with this PR now.

**Comment by jstockwin** [comment]
Hey @pietermarsman and @fakabbir. I've been sorting through the issues (as per conversation on gitter). I've marked the issue relating to this PR as "in progress". Is this moving forwards somehow?

**Comment by pietermarsman** [comment]
> I felt it's getting assigned as `UniGB-UCS2-H` instead of `unknown`. But if that not the case, I wonder why solved the issue. Let me try debug the test case.

That's indeed the case. The differences: 

- Earlier only cmap names from `IDENTITY_ENCODER` were allowed. That behavior was introduced in fa400431f571de9197ab72ffc5dd9f7d76826474, created by @fakabbir, reviewed by me. 
- Now it gets the cmap name from the font specification, uses `IDENTITY_ENCODER` to map some values, and then tries to get the appropriate `CMap` with `CMap.get_cmap()`. If that succeeds it returns that one, if it fails it returns a dummy.

So I think this is an improvement. But it would be great if someone else can confirm this.

**Comment by pietermarsman** [comment]
It improves the output of the issue, so thats a good start :)

**Comment by dwalton76** [comment]
Does the current patch resolve the issue with the Chinese characters in the PDF in #391 ?

**Comment by pietermarsman** [comment]
> Does the current patch resolve the issue with the Chinese characters in the PDF in #391 ?

Yes

**Comment by pietermarsman** [comment]
I need another review before merging this. Either from @fakabbir or someone else.

**Comment by jstockwin** [comment]
I can take a look if you want, but I think fakabbir probably understood this more than I will so you should probably wait for him. Looks like CI is failing at the moment anyway?

**Comment by pietermarsman** [comment]
I thought I fixed that earlier, but actually it never passed :thinking: 

Now it is! :) Actually, the test output improved because the CJK characters in simple3.pdf are now also recognized.

**Comment by pietermarsman** [comment]
Now it is...

File: pdfminer/pdffont.py [hunk 1]
```diff
@@ -134,15 +133,11 @@ def do_keyword(self, pos, token):
...
-# (induced from bug report/PR)
-IDENTITY_ENCODER = {'Identity-H': 'Identity-H',
-                    'Identity-V': 'Identity-V',
-                    'DLIdent-H': 'Identity-H',
-                    'DLIdent-V': 'Identity-V',
-                    'OneByteIdentityH': 'OneByteIdentityH',
-                    'OneByteIdentityV': 'OneByteIdentityV',
-                    }
+# it is harmless and might be used by some PDFs.
+IDENTITY_ENCODER = {
```
Comment (jstockwin) [inline pdfminer/pdffont.py:137]: I'd maybe add a comment to briefly say what this does - i.e. "Used to fix incorrect identities. Identities in this list will be mapped to the given values when getting the cmap".

**Pull request merged**